wordpress-mobile/WordPress-iOS
# [Stats Insights] Load Insights asynchronously

Fixes #12146

This PR enables the async loading block in Insights.

![insights-async](https://user-images.githubusercontent.com/912252/63778446-b49c1900-c8dc-11e9-8cfb-800ba2809741.gif)

At the moment the app displays the _No Result View_ until the store finishes to load the whole overview. This can take some times and block the user experience.
With the async block loading every table section is now independently handled based on a _row status_ (`StoreFetchingStatus`). If the row status is _loading_ and there's no cached data for that section a _Ghost view_ is displayed instead.

The _No Result View_ will be displayed only if there's no cached data and all the API calls will fail.

Also the chart view has priority in the loading queue. This means the other sections will be loaded as soon the LPS is loaded.

This feature is behind the feature flag `statsAsyncLoading`.

## To test:
- To test the async loading you should open a site that doesn't have any stored data
- Also use a link conditioner to simulate a slow connection
- In `FeatureFlag` return false in `case .statsAsyncLoading:` to test everything works fine as before
- If you want to see all the ghost views enabled, return `return rowBlock()` in `SiteStatsInsightsViewModel`
- Test with iOS 13 and Dark mode to check the new colors

Update release notes:
